03_python_fastapi_project/main.py

```
@@ -126,6 +141,101 @@ async def delete_product(product_id: int, db: AsyncSession = Depends(get_db)):
+    cart_items = result.scalars().all()
+    
+    # Load the product relationship manually
+    for item in cart_items:
+        product_result = await db.execute(select(Product).filter(Product.id == item.product_id))
+        item.product = product_result.scalar_one()
+    
+    return cart_items
+
+
+@app.post("/cart", response_model=CartItemDTO, status_code=201)
+async def add_to_cart(cart_item: CartItemCreate, db: AsyncSession = Depends(get_db)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)